src/datasource.ts

```diff
@@ -954,7 +954,12 @@ export class OpenSearchDatasource extends DataSourceWithBackend<OpenSearchQuery,
       }
 
       if (parsedQuery.find === 'terms') {
-        parsedQuery.field = getTemplateSrv().replace(parsedQuery.field, {}, 'lucene');
+        if (parsedQuery.field) {
+          parsedQuery.field = getTemplateSrv().replace(parsedQuery.field, {}, 'lucene');
+        }
+        if (parsedQuery.script) {
+          parsedQuery.script = getTemplateSrv().replace(parsedQuery.script);
```

Comment:
Hello, yes it works as expected.

I thought not to put "lucene" formatting because that field is not actually a Lucene query, is a Painless scripting language field: https://opensearch.org/docs/latest/api-reference/script-apis/exec-script/